cloud/aws-functions/orphaned_eks_clusters.py

```diff
@@ -58,29 +58,41 @@ def wait_for_node_group_delete(autoscaling_client, autoscaling_group_name):
     else:
         logging.error(f"Node group {autoscaling_group_name} was not deleted in {timeout} seconds.")
 
-def delete_nodegroup(aws_region, cluster_name):
+def delete_nodegroup(aws_region):
     autoscaling_client = boto3.client('autoscaling', region_name=aws_region)
-    ec2 = boto3.resource('ec2')
-
-    autoscaling_group_name = ""
-    for instance in ec2.instances.all():
-        tags = instance.tags
-        tags_dict = {item['Key']: item['Value'] for item in tags}
-        if cluster_name not in tags_dict['Name']:
-            continue
-        state = instance.state['Name']
-        if tags_dict['alpha.eksctl.io/cluster-name'] and tags_dict[
-            'alpha.eksctl.io/cluster-name'] == cluster_name and state == 'running':
-            autoscaling_group_name = tags_dict['aws:autoscaling:groupName']
-            break
-        else:
-            continue
-    if len(autoscaling_group_name) != 0:
-        try:
-            autoscaling_client.delete_auto_scaling_group(AutoScalingGroupName=autoscaling_group_name, ForceDelete=True)
-            wait_for_node_group_delete(autoscaling_client, autoscaling_group_name)
-        except Boto3Error as e:
-            logging.error(f"Deleting autoscaling group {autoscaling_group_name} failed with error: {e}")
+    auto_scaling_groups = []
+
+    paginator = autoscaling_client.get_paginator('describe_auto_scaling_groups')
+    for page in paginator.paginate():
+        for group in page['AutoScalingGroups']:
+            tags = group.get('Tags', [])
+            team_value = False
+            delete_cluster = False
+
+            for tag in tags:
+                if (tag['Key']=='team' and tag['Value']=='cloud'):
+                    team_value = True
+
+                if tag['Key'] == 'delete-cluster-after-hours':
+                    delete_cluster = True
+                    cluster_lifetime = float(tag['Value'])
```

Comment:
this will throw exception if value is empty and execution will stop and remaining clusters won't be deleted

Reply:
it'd be better to log the cluster name and continue